libcontainer/factory_linux.go

```diff
@@ -239,19 +239,23 @@ func (l *LinuxFactory) StartInitialization() (err error) {
 		}
 		*pair.v = i
 	}
+
 	var (
 		pipe = os.NewFile(uintptr(pipefd), "pipe")
 		it   = initType(os.Getenv("_LIBCONTAINER_INITTYPE"))
 	)
+
+	// ensure that this pipe is always closed in the case of failure.
+	defer pipe.Close()
```

Comment:
With the updated version I think this is the only outstanding point of contention, LMK your verdict.